plugins/ecc-vscc/ecc_validation_logic.go

```diff
@@ -233,12 +246,25 @@ func (vscc *VSCCECC) checkEnclaveEndorsement(cis *peer.ChaincodeInvocationSpec,
 		if !isValid {
 			return fmt.Errorf("Response invalid! Signature verification failed!")
 		}
+		logger.Debug("Enclave signature validation successfully passed")
 
 	}
 
 	return nil
 }
 
+func isFPCNameSpace(nameSpace string) bool {
+
+	if nameSpace == "_lifecycle" {
+		// do nothing for lifecycle namespace
+		return false
+	}
+
+	// TODO do some real check if namespace is a FPC namespce.
```

Comment:
Can't we solve it right now?  Couldn't we use presence of the "ecc-vscc" validation plugins?  Otherwise, let's revert to the ecc_ (or better, fpc_) prefixing of chaincode name we did earlier? (it's really ugly, but short of changing chaincode definition which we are unlikely going to do for MVP, i guess there are only more or less ugly solutions?  (Personally, the validation-plugin route seems nicer than the prefix one ...)

BTW: this issue exists also in TLCC. So if we do not address it here then we definitely need an issue and work-plan update for it!

Reply:
From discussion on today's call i guess conclusion is that (a) presence of escc-validation is proper way of identification of FPC chaincode, (b) until ERCC is refactored, we need though a workaround for the current erccc cc2cc (presumably based on the "ercc" namespace?) and (c) we need a new (MVP) issue to implement that strategy in TLCC and remove the workaround as soon as ercc is refactored?

Reply:
let me look into that if we can solve it right now or not. Since we are not supporting cc2cc calls with FPC there is a chance that non-fpc chaincodes wont show up in the read/writeset. Only _lifecycle and maybe ercc? but we can exclude them. Will look into that.

Reply:
You mean "non-fpc chaincodes" _except_ ercc?!  In any case, from your latest version i think you consider it resolved for the ecc (and ercc) validation plugins?    I would tend to agree (although see some comments in my review). However, what about correspondingly also fix tlcc: do we have a corresponding issue for that?